patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -757,14 +767,23 @@
 +    }
 +
 +    /**
-+     *
++     * @deprecated use {@link #isWood(IBlockState, IBlockAccess, BlockPos)}
++     */
++    @Deprecated
++    public boolean isWood(IBlockAccess world, BlockPos pos)
++    {
++        return false;
++    }
++
++    /**
++     * @param state The current state
 +     * @param world The current world
 +     * @param pos Block position in world
 +     * @return  true if the block is wood (logs)
 +     */
-+    public boolean isWood(IBlockAccess world, BlockPos pos)
++    public boolean isWood(IBlockState state, IBlockAccess world, BlockPos pos)
```

Comment:
Why does this even exist? It should be done with oredict instead.

Reply:
What? OreDict is for ItemStacks, this is a Block... I'm pretty sure you know the difference and how inefficient it would be to check OreDict every time this is called...

Reply:
@LatvianModder https://github.com/Railcraft/Railcraft/blob/aba7703d2480073998c5a8a02ad2802362fefaba/src/main/java/mods/railcraft/common/plugins/forge/OreDictPlugin.java#L84 Ore dictionary can be used to find if a block state is an ore. What's wrong with that intention? Anyways, another pull can be opened to address this issue.

Reply:
@LatvianModder @liach You do *not* have to search through all the oredict items; you only need to search the ores that the block state matches, which are far fewer in number:
```java
public static boolean isOre(IBlockState state, String ore) {
  return ArrayUtils.contains(OreDictionary.getOreIDs(new ItemStack(state.getBlock(), 1, state.getBlock().damageDropped(state))), OreDictionary.getOreID(ore));
}
```

@liach Your function fails to handle block ores with metadata.

Reply:
Oredict is not useful for blocks which don't have a direct item version.
As an example, something like "burnt wood", which drops "ash" when broken.

Functions like this are often implemented to allow modders to have blocks that can opt-in to behaviour X without also forcefully being opted-in to behaviour Y as well.

Reply:
@Landmaster That is a random bad example, I know to use `Block#getStateFromMeta`

Reply:
Or see #3977 for that